### ISSUE TITLE
Custom registry is sensitive to trailing "/"

### ISSUE THREAD

**Issue by kremerd** [rogeriochaves/npm-force-resolutions#61]
## Summary

NPM can be configured to use a custom registry to resolve packages. `npm-force-resolutions` respects this setting, but requires it to not contain a trailing slash. If it does, the custom registry is silently overriden. This is misleading at best, and might even cause security issues in the right environment.

## How to reproduce

1. Create a new NPM project:

   ```sh
   mkdir test
   cd test
   npm init -y
   ```

2. Create an `.npmrc` file that points to a custom NPM registry. It should look something like this:

   ```properties
   registry = https://my.domain/repository/npm/
   email = test@user.com
   engine-strict = true
   ```

   Make sure there's a **trailing slash** in the registry configuration. While that's not necessary, it's also not wrong, and even used in the default setting according to the [docs](https://docs.npmjs.com/cli/v8/using-npm/config#registry).

3. Install some dependencies and use `npm-force-resolutions` to pin a transitive dependency.

   ```sh
   npm i gulp@4.0.0 -E
   sed -i 's/"ISC",/"ISC",\n  "resolutions": {\n    "glob-parent": "5.1.2"\n  },/' package.json
   # sed -i '' 's/"ISC",/"ISC",\n  "resolutions": {\n    "glob-parent": "5.1.2"\n  },/' package.json # on OS X
   npx npm-force-resolutions
   ```

## Expected Result

All dependencies are retrieved from the custom NPM registry.

## Actual Result

The pinned transitive dependencies are retrieved from `npmjs.org`.

```sh
grep 'npmjs.org' package-lock.json
#          "resolved": "https://registry.npmjs.org/glob-parent/-/glob-parent-5.1.2.tgz",
#      "resolved": "https://registry.npmjs.org/glob-parent/-/glob-parent-5.1.2.tgz",
#          "resolved": "https://registry.npmjs.org/glob-parent/-/glob-parent-5.1.2.tgz",
```

**Comment by hcz** [comment]
Package `npm-force-resolutions` behaves correctly.
This problem is totally unrelated to the trailing slash.

This package makes npm registry api calls and uses tarball info:
```bash
$ curl -s https://my.domain/repository/npm/glob-parent/5.1.2  | jq '.dist.tarball'
"https://registry.npmjs.org/glob-parent/-/glob-parent-5.1.2.tgz"
```

So the problem is probably on your custom registry side.
I had the same problem. Nexus was serving wrong npm package urls.
The best solution from my collegue was to [set up Nginx using replacement rules](http://nginx.org/en/docs/http/ngx_http_sub_module.html) like this:
```
sub_filter 'https://registry.npmjs.org' 'https://my.domain/repository/npm';
sub_filter_once off;
sub_filter_types *;
```

**Comment by kremerd** [comment]
First of all: I've moved away from `npm-force-resolutions`, so this issue no longer affects me. Since nobody else seems to be affected by this, I'm happy to drop my case ;).

Anyway, I don't think your explanation is correct:

1. Yes, the `.dist.tarball` property is pointing to the `npmjs.org` package registry. It does so for all packages that I tried, including `gulp@4.0.0` and `vinyl-fs@3.0.3` (a direct dependency of `gulp`). But those are handled correctly, i.e. the `package-lock.json` references them from my custom registry. I believe this is the case because these parts of the file were written by `npm` directly rather than by `npm-force-resolutions`.
2. The trailing slash _does_ make a difference. If I set `registry = https://my.domain/repository/npm`, everything works fine and in the example above `glob-parent-5.1.2` is resolved from my custom registry as well.
